sqlglot/dialects/hive.py

```diff
@@ -487,8 +487,10 @@ class Generator(generator.Generator):
             exp.If: if_sql(),
             exp.ILike: no_ilike_sql,
             exp.IsNan: rename_func("ISNAN"),
-            exp.JSONExtract: rename_func("GET_JSON_OBJECT"),
-            exp.JSONExtractScalar: rename_func("GET_JSON_OBJECT"),
+            exp.JSONExtract: lambda self, e: self.func("GET_JSON_OBJECT", e.this, e.expression),
+            exp.JSONExtractScalar: lambda self, e: self.func(
+                "GET_JSON_OBJECT", e.this, e.expression
```

Comment:
Btw @tobymao changes like this is why I kinda wanted to filter out bools

Reply:
maybe we can try it later